lib/ts/recipe/accountlinking/recipe.ts

```diff
@@ -275,31 +275,15 @@ export default class Recipe extends RecipeModule {
         recipeUserId: string;
         userContext: any;
     }): Promise<string> => {
-        let accountInfo: AccountInfo;
-        if (newUser.email !== undefined) {
-            accountInfo = {
-                email: newUser.email,
-            };
-        } else if (newUser.phoneNumber !== undefined) {
-            accountInfo = {
-                phoneNumber: newUser.phoneNumber,
-            };
-        } else if (newUser.thirdParty !== undefined) {
-            accountInfo = {
-                thirdPartyId: newUser.thirdParty.id,
-                thirdPartyUserId: newUser.thirdParty.userId,
-            };
-        } else {
-            throw Error("this error should never be thrown");
-        }
-        let users = await this.recipeInterfaceImpl.listUsersByAccountInfo({
-            accountInfo,
+        let user = await this.getPrimaryUserIdThatCanBeLinkedToRecipeUserId({
+            recipeUserId,
             userContext,
         });
```

Comment:
be sure that this is correct

Reply:
This is correct cause this does the same thing that the diagram says, but in addition, it also checks the account to link table.